server/src/test/java/org/elasticsearch/search/aggregations/metrics/tophits/TopHitsAggregatorTests.java

```diff
@@ -199,4 +206,50 @@ public void testSetScorer() throws Exception {
         reader.close();
         directory.close();
     }
+
+    public void testTopHitsMaxResultWindow() throws Exception {
+        int maxResultWindow = randomIntBetween(5, 10);
+        IndexSettings indexSettings = new IndexSettings(
+            IndexMetaData.builder("_index")
+                .settings(Settings.builder().put(IndexMetaData.SETTING_VERSION_CREATED, Version.CURRENT))
+                .numberOfShards(1)
+                .numberOfReplicas(0)
+                .creationDate(System.currentTimeMillis())
+                .build(),
+            Settings.builder().put("index.max_result_window", maxResultWindow).build()
+        );
+        try (Directory directory = newDirectory();
+             IndexWriter w = new IndexWriter(directory, newIndexWriterConfig());
+             IndexReader reader = DirectoryReader.open(w)) {
+            IndexSearcher searcher = new IndexSearcher(reader);
+            MultiBucketConsumerService.MultiBucketConsumer consumer = new MultiBucketConsumerService.MultiBucketConsumer(DEFAULT_MAX_BUCKETS);
+            // from + size > maxResultWindow
+            int from = randomInt(10);
+            int size = maxResultWindow - from + randomIntBetween(1, 10);
+            String name = randomAlphaOfLength(10);
+            IllegalArgumentException ex = expectThrows(IllegalArgumentException.class,
+                () -> createAggregatorFactory(
+                    new TopHitsAggregationBuilder(name).from(from).size(size),
+                    searcher,
+                    indexSettings,
+                    consumer));
+            assertEquals("Top hits result window is too large, the top hits aggregator [" +
+                name + "]'s from + size must be less " +
+                "than or equal to: [" + maxResultWindow + "] but was [" + (from + size) +
+                "]. This limit can be set by changing the [" + IndexSettings.MAX_RESULT_WINDOW_SETTING.getKey() +
+                "] index level setting.", ex.getMessage());
+
+            // from + size <= maxResultWindow
+            int from2 = randomIntBetween(1, maxResultWindow - 5);
+            int size2 = randomIntBetween(1, maxResultWindow - from);
```

Comment:
I think "from" should be "from2" here. Also I would make the calculation simpler here as well, I think otherwise min might be smaller than max in the calls to "randomIntBetween"

Reply:
Thansk @cbuescher , I pushed 803e3f6757220d76c79b8f881ae005903378f0df